### PR TITLE
Homework 29

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/inject/InjectDataController.java [hunk 1]
```diff
@@ -31,33 +33,25 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+        User admin = new User(
+                "admin",
+                "admin",
+                "jason@ukr.net",
+                "admin",
+                "2");
+        admin.setRoles(Set.of(Role.of("ADMIN")));
+        userService.create(admin);
+        shoppingCartService.create(new ShoppingCart(1L));
+        shoppingCartService.create(new ShoppingCart(2L));
```
Comment (Nokafan) [inline src/main/java/com/internet/shop/controllers/inject/InjectDataController.java]: Maybe better to use id from the user, not like the magic number.

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        if (protectedUrls.get(reqUrl) == null) {
+            chain.doFilter(req, resp);
+            return;
+        }
+
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        User user = userService.getById(userId);
+        if (isAuthorized(user, protectedUrls.get(reqUrl))) {
+
```
Comment (voloshynamaryna11) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java]: empty line

**Review by voloshynamaryna11 — COMMENTED** [review]
cool

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java]: done

File: src/main/java/com/internet/shop/controllers/inject/InjectDataController.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/inject/InjectDataController.java]: agree

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/orders", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/products", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/products/delete", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/product/add", Set.of(Role.RoleName.USER));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java:36]:
```suggestion
        protectedUrls.put("/shopping-cart/products/delete", Set.of(Role.RoleName.USER));
        protectedUrls.put("/shopping-cart/products/add", Set.of(Role.RoleName.USER));
```

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job!

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java:36]: done

**Pull request merged**